File: common/src/main/kotlin/org/virtuslab/inkuire/kotlin/model/OutputFormat.kt
package org.virtuslab.inkuire.kotlin.model

data class Match(
    val prettifiedSignature: String,
    val functionName: String,
    val packageLocation: String,
    val pageLocation: String
)

data class OutputFormat(
    val query: String,
    val matches: List<Match>
)

File: engine/shared/src/main/scala/org/virtuslab/inkuire/engine/impl/model/OutputFormat.scala
package org.virtuslab.inkuire.engine.impl.model

case class Match(
  prettifiedSignature: String,
  functionName:        String,
  packageLocation:     String,
  pageLocation:        String,
  entryType:           String,
  mq:                  Int
)

sealed trait OutputFormat

case class ResultFormat(
  query:   String,
  matches: List[Match]
) extends OutputFormat

case object EndFormat extends OutputFormat
